examples/doublemach-mpi.py

```diff
@@ -0,0 +1,491 @@
+        PrescribedFluidBoundary(boundary_state_func=_boundary_state),
+        DTAG_BOUNDARY("ic3"):
+        PrescribedFluidBoundary(boundary_state_func=_boundary_state),
+        DTAG_BOUNDARY("wall"): AdiabaticNoslipMovingBoundary(),
+        DTAG_BOUNDARY("out"): AdiabaticNoslipMovingBoundary(),
+    }
+
+    if rst_filename:
+        current_t = restart_data["t"]
+        current_step = restart_data["step"]
+        current_cv = restart_data["cv"]
+        if logmgr:
+            from mirgecom.logging_quantities import logmgr_set_time
+            logmgr_set_time(logmgr, current_step, current_t)
+    else:
+        # Set the current state from time 0
+        current_cv = initializer(nodes)
+    current_state = make_fluid_state(cv=current_cv, gas_model=gas_model)
+
+    visualizer = make_visualizer(discr,
+                                 discr.order if discr.dim == 2 else discr.order)
+
+    initname = initializer.__class__.__name__
+    eosname = eos.__class__.__name__
+    init_message = make_init_message(
+        dim=dim,
+        order=order,
+        nelements=local_nelements,
+        global_nelements=global_nelements,
+        dt=current_dt,
+        t_final=t_final,
+        nstatus=nstatus,
+        nviz=nviz,
+        cfl=current_cfl,
+        constant_cfl=constant_cfl,
+        initname=initname,
+        eosname=eosname,
+        casename=casename,
+    )
+    if rank == 0:
+        logger.info(init_message)
+
+    def my_write_viz(step, t, state, dv, tagged_cells):
+        viz_fields = [("cv", state),
+                      ("dv", dv),
+                      ("tagged_cells", tagged_cells)]
+        from mirgecom.simutil import write_visfile
+        write_visfile(discr, viz_fields, visualizer, vizname=casename,
+                      step=step, t=t, overwrite=True)
+
+    def my_write_restart(step, t, state):
+        rst_fname = rst_pattern.format(cname=casename, step=step, rank=rank)
+        if rst_fname != rst_filename:
+            rst_data = {
+                "local_mesh": local_mesh,
+                "cv": state,
+                "t": t,
+                "step": step,
+                "order": order,
+                "global_nelements": global_nelements,
+                "num_parts": nparts
+            }
+            from mirgecom.restart import write_restart_file
+            write_restart_file(actx, rst_data, rst_fname, comm)
+
+    def my_health_check(state, dv):
+        # Note: This health check is tuned s.t. it is a test that
+        #       the case gets the expected solution.  If dt,t_final or
+        #       other run parameters are changed, this check should
+        #       be changed accordingly.
+        health_error = False
+        from mirgecom.simutil import check_naninf_local, check_range_local
+        if check_naninf_local(discr, "vol", dv.pressure):
+            health_error = True
+            logger.info(f"{rank=}: NANs/Infs in pressure data.")
+
+        from mirgecom.simutil import allsync
```

Comment:
I believe these should be `global_reduce` now

Reply:
[7b9ee93](https://github.com/illinois-ceesd/mirgecom/pull/582/commits/7b9ee9348c70a3424ac40369b942790bf4524aa8)